com.unity.netcode.gameobjects/Tests/Runtime/NetworkVariableTests.cs

```diff
@@ -4,8 +4,63 @@
 using UnityEngine.TestTools;
 using NUnit.Framework;
 
+
+
 namespace Unity.Netcode.RuntimeTests
 {
+    public unsafe struct FixedStringStruct : INetworkSerializable
```

Comment:
My overall question when it comes to fixed strings is... what's the use case? Any time you need to actually write a string to a log or pass it to a function or draw it to the screen... wouldn't you end up needing to convert it to string and thus incurring an allocation? So does it actually stop allocations or does it just delay them?

One thing I was going to experiment with was doing something like this:

```c#
struct FixedString
{
  public readonly string String;

  public FixedString(int size)
  {
    String = "".PadRight(size);
  }

  public string AsString()
  {
    return String;
  }

  public unsafe char this[int index]
  {
    get => m_Value[index];
    unsafe set
    {
      fixed(char* charPtr = String)
      {
        charPtr[index] = value;
      }
    }
  }
}
```

That would give you a mutable string - one-time allocation up front, but from there you're able to overwrite the memory of the string directly instead of creating new ones.

Reply:
This sketch came up because someone was asking how we do strings with NetworkVariables now that we restrict them to unmanaged types.  The answer is, especially as these vars get snapshotted and as we move toward "performant netcode / MTU-friendly oriented", we do not want arbitrary length types in netvars.  Yet I was interested in seeing what would happen if we used a fixed string scheme.  In that sense the pattern with a struct with a byte/char array of declared size is more what I'm trying to test / demo

Reply:
Oooo... I like your approach Jaedyn.  It only unsafely encapsulates when you are setting the value directly.  You padright call is a nice touch... to make sure it ends it properly.

Reply:
@ShadauxCat 
The fixed buffer allows you to re-use an already allocated string.
So, if you were to have like a player name that could vary in size then you could have something like this defined in a class as a NetworkVariable<T> which would allow you to have a "placeholder" string that wasn't "nullable".